dipy/tracking/local/tests/test_local_tracking.py

```diff
@@ -158,6 +162,27 @@ def get_direction(self, p, d):
     bad_affine[0, 1] = 1.
     npt.assert_raises(ValueError, LocalTracking, dg, tc, seeds, bad_affine, 1.)
 
+    # Sometimes data have affines with tiny shear components.
+    # For example, the small_101D data-set has some of that:
+    fdata, fbval, fbvec = dpd.get_data('small_101D')
+    funky_affine = nib.load(fdata).get_affine()
+
+    # We still want to be able to track with this kind of data:
+    streamlines_funky = LocalTracking(direction_getter=dg,
+                                      tissue_classifier=tc,
+                                      seeds=seeds,
+                                      affine=funky_affine,
+                                      step_size=1.,
+                                      return_all=True)
+
+    streamlines_funky = iter(streamlines_funky)
+    # We do the same checks here:
+    y = 0
+    sl = next(streamlines_funky)
+    npt.assert_equal(sl[0], [0, y, 0])
```

Comment:
to get the same results with a different affine, you need to apply the inverse transform to the seed positions, then the inverse affine transformation the the generated streamlines points. The step size must also be adjusted to the voxel size to obtain the same tracking points. 

We miss testing for various affine cases here.. I can jump into it if you want.